R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,447 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
+      L2322.StubTechFixOut_Fert_imp
+
+    # L2322.StubTechFixOut_Fert_exp: fixed output of import technology (fixed imports)
+    # Exports are positive net exports
+    L142.ag_Fert_NetExp_MtN_R_Y %>%
+      filter(year %in% BASE_YEARS) %>%
+      rename(fixedOutput = value) %>%
+      mutate(fixedOutput = round(fixedOutput, energy.DIGITS_CALOUTPUT)) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID" ) %>%
+      mutate(supplysector = filter(A322.globaltech_shrwt, grepl( "Exports", supplysector))[["supplysector"]]) %>%
+      mutate(subsector = filter(A322.globaltech_shrwt, grepl( "Exports", supplysector))[["subsector"]]) %>%
+      mutate(stub.technology = filter(A322.globaltech_shrwt, grepl( "Exports", supplysector))[["technology"]]) %>%
+      mutate(fixedOutput = pmax(0, fixedOutput), share.weight.year = year, subs.share.weight = 0, tech.share.weight = 0) ->
+      L2322.StubTechFixOut_Fert_exp_base
+
+    L2322.StubTechFixOut_Fert_exp_base %>%
+      bind_rows(repeat_add_columns(select(filter(., year == max(BASE_YEARS)), -year), tibble(year = FUTURE_YEARS))) %>% # Repeat final year to all future years and rbind
+      select(LEVEL2_DATA_NAMES[["StubTechFixOut"]]) ->
+      L2322.StubTechFixOut_Fert_exp
+
+    # L2322.PerCapitaBased_Fert: per-capita based flag for fertilizer exports final demand
+    tibble(region = GCAM_region_names[["region"]]) %>%
+      mutate(energy.final.demand = filter(A322.globaltech_shrwt, grepl( "Exports", supplysector))[["supplysector"]]) %>%
+      mutate(perCapitaBased = 0) ->
+      L2322.PerCapitaBased_Fert
+
+    # L2322.BaseService_Fert: base-year service output of fertilizer exports final demand
+    # Base service is equal to the output of the exports supplysector
+    tibble(region = L2322.StubTechFixOut_Fert_exp_base[["region"]]) %>%
+      mutate(energy.final.demand = L2322.StubTechFixOut_Fert_exp_base[["supplysector"]]) %>%
+      mutate(year = L2322.StubTechFixOut_Fert_exp_base[["year"]]) %>%
+      mutate(base.service = L2322.StubTechFixOut_Fert_exp_base[["fixedOutput"]]) ->
+      L2322.BaseService_Fert
+
+    # ===================================================
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+
+    L2322.Supplysector_Fert %>%
+      add_title("Supply sector information for fertilizer sector") %>%
+      add_units("NA") %>%
+      add_comments("Expanded fertilizer sector supply sector information for all GCAM regions") %>%
```

Comment:
Could you include more details? e.g., what is in this information (is it logits, etc.)? And, where does it come from (i.e., an assumption file)?